machineEnvironment.sh

```diff
@@ -35,6 +35,13 @@ nthreads=""     # number of threads to use for parallel builds
 mpilaunch=""    # command to launch an MPI executable (e.g. aprun)
 installdir=""   # directory where libraries are installed
 
+# set default value for useslurm based on whether a submit script exists
+envdir=`dirname $0`
```

Comment:
I would actually set a variable `scheduler="slurm"` or `scheduler="none"` here which is then used in the `runJob.sh` script. This leaves wiggle room for other schedulers and does not hardcode something SLURM specific.

Reply:
I like that!